Source/Plugins/EditorModules/LogView/Modules/LogView.cs

```diff
@@ -1,26 +1,34 @@
 ﻿using System;
 using System.Drawing;
 using System.Collections.Generic;
-using System.Globalization;
-using System.Linq;
-using System.Text;
-using System.Xml.Linq;
 using System.Windows.Forms;
 
 using WeifenLuo.WinFormsUI.Docking;
 
-using Duality;
-using Duality.Editor;
-
 namespace Duality.Editor.Plugins.LogView
 {
 	public partial class LogView : DockContent
 	{
+		private LogViewSettings userSettings = new LogViewSettings();
 		private int unseenWarnings = 0;
-		private int unseenErrors   = 0;
+		private int unseenErrors = 0;
 		private Dictionary<string,ToolStripButton> sourceFilterButtons = new Dictionary<string,ToolStripButton>();
 
 
+		public LogViewSettings UserSettings
+		{
+			get { return this.userSettings; }
+			set
+			{
+				if (this.userSettings != value)
+				{
+					this.userSettings = value;
+					this.ApplyUserSettings();
+				}
```

Comment:
Spotted a slight issue: Calling `Apply` only with a different instance will break when

- The `LoadUserData` code when it happens as part of a `SettingsContainer.Apply` call in the editor, since it will be the same instance, maybe only with changed field values and also when
- Settings serialization is one day extended to allow updating an existing instance.

So we should either call `ApplyUserSettings` explicitly after assignment in `LoadUserData`, or always apply on assignment. Currently leaning towards the first option, but not sure yet.

Reply:
> The LoadUserData code when it happens as part of a SettingsContainer.Apply call in the editor, since it will be the same instance, maybe only with changed field values and also when

In this case the developer has to call Apply himself. Not really a way around that I think.

> Settings serialization is one day extended to allow updating an existing instance.

If this happens then we don't really need a Settingscontainer anymore as we can just use the setting class directly. That setting class can then implement changed/changing events so we could even make the change events more fine grained.

Reply:
Is this really a issue atm though? Before this refactor I don't think we even had a changed event for these pluginsettings do it not a regression.

In the future when we can just reuse the same instance we might look into implementing `INotifyPropertyChanged` or similar.

Reply:
> In this case the developer has to call Apply himself. Not really a way around that I think.

Yes, but the issue is that even when calling `Apply` on the global editor settings container, it will be ignored by the log view UI. I think having to call it is fine as (like you say) there's not really a great way around that right now - but the second part is just a bug.

It's not a regression, but since we're establishing a new model here, we should at least make sure to remain consistent within that model. We just didn't have this problem before because there were no settings exposed in code whatsoever.

It's nothing major to fix, probably just a few lines after choosing which way to go, so might as well do it 😁 I can do a quick followup commit as soon as I get around to it.